spiffworkflow-backend/src/spiffworkflow_backend/services/assertion_service.py

```diff
@@ -0,0 +1,18 @@
+"""Assertion_service."""
+import contextlib
+from typing import Generator
+
+import sentry_sdk
+from flask import current_app
+
+
+@contextlib.contextmanager
+def safe_assertion(condition: bool) -> Generator[bool, None, None]:
+    try:
+        yield True
+    except AssertionError as e:
+        if not condition:
+            sentry_sdk.capture_exception(e)
+            current_app.logger.exception(e)
```

Comment:
i'm not very confident in this, but sentry seems to override logger.exception in some cases and send the errors to sentry. just pointing this out in case we see doubled errors.

Reply:
Funny thing I saw that when logging the exception as error, which I would have thought would have been ok. But yes will keep an eye on it and switch to info if needed.